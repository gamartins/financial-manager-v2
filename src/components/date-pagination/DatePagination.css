.DatePagination .page-item {
    cursor: pointer;
}
