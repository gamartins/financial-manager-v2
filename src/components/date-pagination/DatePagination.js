import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { format } from 'date-fns'

import './DatePagination.css'

export default class DatePagination extends Component {
  previous() {
    this.props.previous()
  }

  next() {
    this.props.next()
  }

  render() {
    return (
      <nav>
        <ul className="DatePagination pagination">
          <li id="item-previous" className="page-item" onClick={() => this.previous()}>
            <span className="page-link">&laquo;</span>
          </li>
          <li id="item-actual" className="page-item">
            <span className="page-link">
              {format(this.props.date, this.props.format)}
            </span>
          </li>
          <li id="item-next" className="page-item" onClick={() => this.next()}>
            <span className="page-link">&raquo;</span>
          </li>
        </ul>
      </nav>
    )
  }
}

DatePagination.propTypes = {
  date: PropTypes.instanceOf(Date).isRequired,
  format: PropTypes.string.isRequired,
  previous: PropTypes.func,
  next: PropTypes.func,
}
