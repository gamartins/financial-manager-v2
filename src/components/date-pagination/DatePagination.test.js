import React from 'react'
import Enzyme, { mount, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { format } from 'date-fns'

import DatePagination from './DatePagination';

jest.mock('../../services/CategoryService');

Enzyme.configure({ adapter: new Adapter() });
  
describe('<DatePagination />', () => {
  test('should be created', () => {
    const wrapper = shallow(<DatePagination date={new Date()} format={'MMM'} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('should call previous on page-item previous clicked', () => {
    const mockPrevious = jest.fn(() => {})
    const wrapper = shallow(<DatePagination
      date={new Date()}
      format={'MMM'}
      previous={mockPrevious}/>)

    wrapper.find('#item-previous').simulate('click')

    expect(mockPrevious).toHaveBeenCalled()
  })

  test('should call previous on page-item next clicked', () => {
    const mockNext = jest.fn(() => {})
    const wrapper = shallow(<DatePagination
      date={new Date()}
      format={'MMM'}
      next={mockNext}/>)

    wrapper.find('#item-next').simulate('click')

    expect(mockNext).toHaveBeenCalled()
  })

  test('should show as defined in format', () => {
    const date = new Date()
    const patern = 'MMM'
    
    const wrapper = shallow(<DatePagination date={date} format={patern} />)
    
    const formatedDate = format(date, patern)
    const componentFormatedDate = wrapper.find('#item-actual').text()

    expect(componentFormatedDate).toBe(formatedDate)
  })

})