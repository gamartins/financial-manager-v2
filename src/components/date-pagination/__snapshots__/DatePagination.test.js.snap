// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatePagination /> should be created 1`] = `ShallowWrapper {}`;
